connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/EmailRouteBuilder.java

```diff
@@ -0,0 +1,300 @@
+         * simply fetch all the users as normal. If it exists, we just fetch
+         * the users from that particular group.
+         */
+        from(direct(Routes.FETCH_GROUP))
+            .routeId(Routes.FETCH_GROUP)
+            // Clear all the headers that may come from the previous route.
+            .removeHeaders("*")
+            .process(this.rbacGroupRequestPreparerProcessor)
+            .to(this.emailConnectorConfig.getRbacURL())
+            .process(this.rbacGroupProcessor)
+            .choice()
+                .when(simpleF("${exchangeProperty.%s}", RBACConstants.RBAC_GROUP_IS_PLATFORM_DEFAULT))
+                    .to(direct(Routes.FETCH_USERS))
+                .otherwise()
+                    .to(direct(Routes.FETCH_GROUP_USERS))
+            .end()
+            // When a 404 "Not Found" status code is received, then we simply
+            // continue with the execution.
+            .onException(HttpOperationFailedException.class)
+                .onWhen(simpleF("${exception.statusCode} == %d", HttpStatus.SC_NOT_FOUND))
+                    .handled(true)
+                    .setProperty(ExchangeProperty.USERNAMES, constant(new HashSet<String>()));
+
+        /*
+         * Fetch the users from an RBAC group and filters the results according
+         * to the recipient settings.
+         */
+        from(direct(Routes.FETCH_GROUP_USERS))
+            .routeId(Routes.FETCH_GROUP_USERS)
+            // Clear all the headers that may come from the previous route.
+            .removeHeaders("*")
+            // Initialize the offset and the loop condition so that we at least
+            // enter once on it.
+            .setProperty(ExchangeProperty.LIMIT, constant(this.emailConnectorConfig.getRbacElementsPerPage()))
+            .setProperty(ExchangeProperty.OFFSET, constant(0))
+            .loopDoWhile(this.notFinishedFetchingAllPages)
+                .process(this.rbacGroupPrincipalsRequestPreparerProcessor)
+                .to(this.emailConnectorConfig.getRbacURL())
+                .process(this.rbacUsersProcessor)
+            .end()
+            .process(this.recipientsFilter)
+            .to(direct(Routes.SEND_EMAIL_BOP));
+
+        from(direct(Routes.SEND_EMAIL_BOP_CHOICE))
+            .choice()
+                .when(constant(this.emailConnectorConfig.isSingleEmailPerUserEnabled()))
+                    .to(direct(Routes.SEND_EMAIL_BOP_SINGLE_PER_USER))
+                .otherwise()
+                    .to(direct(Routes.SEND_EMAIL_BOP))
+            .end();
+
+        /*
+         * Prepares the payload accepted by BOP and sends the request to
+         * the service.
+         */
+        from(direct(Routes.SEND_EMAIL_BOP))
+            .routeId(Routes.SEND_EMAIL_BOP)
+            // Clear all the headers that may come from the previous route.
+            .removeHeaders("*")
+            .process(this.BOPRequestPreparer)
+            .to(this.emailConnectorConfig.getBopURL())
```

Comment:
In `WebClientProducer` we're building the client for BOP with `trustAll=true`. I'm not sure why (this has been here forever) but we may need to disable the certificates verification while calling BOP in this connector.
https://github.com/RedHatInsights/notifications-backend/blob/2ba4d8c1961796baf4bbfadb8eaba5f17959b94d/engine/src/main/java/com/redhat/cloud/notifications/processors/webclient/WebClientProducer.java#L42

Reply:
We can start testing in stage without `trustAll=true` and see if that works.

Reply:
... and another one for this one.